subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/AbstractIncrementalAnnotationProcessingIntegrationTest.groovy

```diff
@@ -114,4 +114,39 @@ abstract class AbstractIncrementalAnnotationProcessingIntegrationTest extends Ab
         outputs.recompiledClasses("A", "B")
     }
 
+    def "recompiles when a resource is removed"() {
+        given:
+        buildFile << """
+            compileJava.inputs.dir 'src/main/resources'
+        """
+        java("class A {}")
+        java("class B {}")
+        def resource = file("src/main/resources/foo.txt")
+        resource.text = 'foo'
+
+        outputs.snapshot { succeeds 'compileJava' }
+
+        when:
+        resource.delete()
+
+        then:
+        succeeds 'compileJava'
+        outputs.recompiledClasses("A", "B")
+    }
+
+    def "does not recompiles when an empty package is added"() {
+        given:
+        java("class A {}")
+        java("class B {}")
+
+        outputs.snapshot { succeeds 'compileJava' }
+
+        when:
+        file('src/main/java/different').createDir()
+
+        then:
+        succeeds 'compileJava'
```

Comment:
❌ I wonder if we need to check here that the task actually executed. If we at some point in time decide to ignore empty directories in the java source directory, then this test would pass even if the underlying code is broken.

Reply:
The user cares about doing an incremental compile when an empty directory is added an another file is modified. That is probably what we should test, and not that nothing is recompiled when an empty directory is added.

Reply:
Test adjusted.